muskathlon/models/survey_user_input.py

```diff
@@ -31,12 +31,19 @@ def write(self, vals):
                     'muskathlon.stage_fill_profile').id)
             ])
 
+            survey = registrations.event_id.medical_survey_id
+            medical_survey_id = self.env['survey.user_input'].search([
+                ('partner_id', '=', registrations.partner_id_id),
```

Comment:
Here you will need to loop over the registrations in order to assign the correct medical survey for each registration.